src/manifolds/StiefelSubmersionMetric.jl

```diff
@@ -0,0 +1,290 @@
+@doc raw"""
+    StiefelSubmersionMetric{T<:Real} <: RiemannianMetric
+
+The submersion (or normal) metric family on the [`Stiefel`](@ref) manifold.
+
+# Constructor
+
+    StiefelSubmersionMetric(α)
+
+Construct the submersion metric on the Stiefel manifold with the parameter ``α > -1``.
+
+The submersion metric family has two special cases:
+- ``α = -\frac{1}{2}``: [`EuclideanMetric`](@ref)
+- ``α = 0``: [`CanonicalMetric`](@ref)
+"""
+struct StiefelSubmersionMetric{T<:Real} <: RiemannianMetric
+    α::T
+end
+
+struct StiefelShootingInverseRetraction{
```

Comment:
We could omit the Stiefel here, since that could be used to arbitrary Shooting methods as well?

Reply:
Maybe! I'll need to look at the original paper. There are many ways to implement shooting. Many of them user derivatives. This one avoids derivatives by using vector transport. To simply call it `ShootingInverseRetraction` we may want to implement something more generic.

Reply:
Or even just reusing the type later

Reply:
Okay I made it a generic method and then specialized it for `k <= div(n, 2)`, so we could remove the `k`-shooting inverse retraction type.